src/Security/Authentication/OpenIdConnect/src/OpenIdConnectConfigureOptions.cs

```diff
@@ -0,0 +1,136 @@
+        options.ReturnUrlParameter = configSection[nameof(options.ReturnUrlParameter)] ?? options.ReturnUrlParameter;
+        options.SaveTokens = StringHelpers.ParseValueOrDefault(options.SaveTokens, configSection[nameof(options.SaveTokens)], bool.Parse);
+        var scopes = configSection.GetSection(nameof(options.Scope)).GetChildren().Select(scope => scope.Value).OfType<string>();
+        foreach (var scope in scopes)
+        {
+            options.Scope.Add(scope);
+        }
+        options.SignedOutCallbackPath = new PathString(configSection[nameof(options.SignedOutCallbackPath)] ?? options.SignedOutCallbackPath);
+        options.SignedOutRedirectUri = configSection[nameof(options.SignedOutRedirectUri)] ?? options.SignedOutRedirectUri;
+        options.SignInScheme = configSection[nameof(options.SignInScheme)];
+        options.SignOutScheme = configSection[nameof(options.SignOutScheme)];
+        options.SkipUnrecognizedRequests = StringHelpers.ParseValueOrDefault(options.SkipUnrecognizedRequests, configSection[nameof(options.SkipUnrecognizedRequests)], bool.Parse);
+        options.UsePkce = StringHelpers.ParseValueOrDefault(options.UsePkce, configSection[nameof(options.UsePkce)], bool.Parse);
+        options.UseTokenLifetime = StringHelpers.ParseValueOrDefault(options.UseTokenLifetime, configSection[nameof(options.UseTokenLifetime)], bool.Parse);
+    }
+
+    private static CookieBuilder GetCookieFromConfig(IConfiguration cookieConfigSection, CookieBuilder cookieBuilder)
+    {
+        if (cookieConfigSection is not null && cookieConfigSection.GetChildren().Any())
+        {
+            // Override the existing defaults when values are set instead of constructing
+            // an entirely new CookieBuilder.
+            cookieBuilder.Domain = cookieConfigSection[nameof(cookieBuilder.Domain)];
+            cookieBuilder.HttpOnly = StringHelpers.ParseValueOrDefault(cookieBuilder.HttpOnly, cookieConfigSection[nameof(cookieBuilder.HttpOnly)], bool.Parse);
+            cookieBuilder.IsEssential = StringHelpers.ParseValueOrDefault(cookieBuilder.IsEssential, cookieConfigSection[nameof(cookieBuilder.IsEssential)], bool.Parse);
+            cookieBuilder.Expiration = cookieBuilder.Expiration is TimeSpan expiration ? StringHelpers.ParseValueOrDefault(expiration, cookieConfigSection[nameof(cookieBuilder.Expiration)], _invariantTimeSpanParse) : cookieBuilder.Expiration;
+            cookieBuilder.MaxAge = cookieBuilder.MaxAge is TimeSpan maxAge ? StringHelpers.ParseValueOrDefault(maxAge, cookieConfigSection[nameof(cookieBuilder.MaxAge)], _invariantTimeSpanParse) : cookieBuilder.MaxAge;
+            cookieBuilder.Name = cookieConfigSection[nameof(CookieBuilder.Name)] ?? cookieBuilder.Name;
+            cookieBuilder.Path = cookieConfigSection[nameof(CookieBuilder.Path)];
+            cookieBuilder.SameSite = cookieConfigSection[nameof(CookieBuilder.SameSite)]?.ToLowerInvariant() switch
```

Comment:
Is there a reason you didn't use Enum.Parse?

Reply:
It doesn't support case-insensitive parsing as far as I can tell.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.enum.parse?view=net-6.0#system-enum-parse(system-type-system-string-system-boolean)